notification/slack/channel.go

```diff
@@ -55,21 +56,33 @@ type Channel struct {
 	Name string
 }
 
-type slackError string
+type apiError struct {
+	msg    string
+	header http.Header
+}
+
+func (err apiError) Error() string { return err.msg }
+
+func mapError(ctx context.Context, err error) error {
+	var apiError *apiError
+	if !errors.As(err, &apiError) {
+		return err
+	}
 
-func (err slackError) Error() string     { return string(err) }
-func (err slackError) ClientError() bool { return true }
-func wrapError(errMsg, details string) error {
-	switch errMsg {
+	switch apiError.msg {
 	case "missing_scope":
-		// happens if the ID is for a user
-		return validation.NewFieldError("ChannelID", "Only channels supported.")
+		acceptedScopes := apiError.header.Get("X-Accepted-Oauth-Scopes")
+		providedScopes := apiError.header.Get("X-Oauth-Scopes")
+		log.Log(ctx, fmt.Errorf("Slack: missing_scope; need one of %v but got %v", acceptedScopes, providedScopes))
```

Comment:
It may be useful to include the context of the original error and then just add the information to the end:

```suggestion
		log.Log(ctx, fmt.Errorf("%w; need one of %v but got %v", err, acceptedScopes, providedScopes))
```

This would read something like: `lookup Slack channel: missing_scope; need one of ...` so we know where it came from before hitting `mapError`

Reply:
An alternative that might be cleaner could be to put the logic in apiError

```go
func (err apiError) Error() string{
  if err.msg == "missing_scope" {
    // accepted/provided from err.header
    return fmt.Sprintf("missing_scope; need one of %v but got %v", acceptedScopes, providedScopes)
}

// in `mapError`
switch apiError.msg {
// can group all permission denied errors this way
case "missing_scope", "token_revoked", "etc...":
  log.Log(ctx, err)
  return validation.NewError("Permission Denied")
}
```